### PR TITLE
[MCC-412658] Break up large client file

### PR DESCRIPTION
**Pull request by cmcinnes-mdsol**
@mdsol/team-16 

This PR breaks up the large `client.rb` file. No functional changes have been made. Opening as draft for now.

So far I've moved `Authenticator`, `LocalAuthenticator`, `Signer`, `RemoteAuthenticator`, and the error classes in to their own files. I also put the `v2.rb` methods back in their respective modules and stopped the monkey patching in that file.

### CODE REVIEW

**Comment by cmcinnes-mdsol** [comment]
@mdsol/team-16 the file structure:

BEFORE REORG
* lib/mauth
  * client.rb
* spec
  * mauth_client_spec.rb


AFTER REORG 

* lib/mauth
  * client.rb 
  * errors.rb
  * client 
    * authenticator_base.rb
    * local_authenticator.rb
    * remote_authenticator.rb 
    * security_token_cacher.rb 
    * signer.rb
* spec
  * mauth_client_spec.rb
  * client
    * local_authenticator_spec.rb
    * remote_authenticator_spec.rb
    * signer_spec.rb
  * support
    * shared_contexts 
      * client.rb 
      * authenticator_base.rb

File: lib/mauth/client.rb [hunk 1]
```diff
@@ -144,6 +120,11 @@ class Client
     class ConfigurationError < StandardError; end
```
[the remark this answers is not in the document]
Comment (cmcinnes-mdsol) [reply lib/mauth/client.rb]: Moved it over.

**Comment by ejennings-mdsol** [comment]
Can you fix the
```
mauth-client-ruby/spec/signable_spec.rb:16: warning: already initialized constant Dummy::SIGNATURE_COMPONENTS_V2
mauth-client-ruby/spec/signable_spec.rb:16: warning: previous definition of SIGNATURE_COMPONENTS_V2 was here
```
warnings when running the specs?

Also wondering if `mauth_client_spec.rb` should just be `client_spec.rb`.

**Comment by cmcinnes-mdsol** [comment]
@ejennings-mdsol got rid of the warnings and renamed spec file to `client_spec.rb`

**Pull request merged**